test/common/router/router_upstream_log_test.cc

```diff
@@ -411,5 +416,29 @@ name: accesslog
   EXPECT_EQ(output_.front(), "cluster_0");
 }
 
+TEST_F(RouterUpstreamLogTest, OnRequestLog) {
```

Comment:
I'd definitely like an e2e test both of the standard case and a second log for retried request.  waitForAccessLog should work as long as you make sure to use the optional fields to wait for the right log entry.

Reply:
I'm not sure how to cause retry for the router in integration tests. Is there even currently an integration test that checks that the standard upstream log is flushed twice for retry?
Anyways, I did add a check to integration tests, that if the bool config is enabled, then there are two upstream access log made.

Reply:
Never mind, figured it out. I renamed ``upstream_filter_state_integration_test.cc`` to ``upstream_access_log_integration_test.cc`` since it only had a single test, that was tightly coupled with upstream access logs, and added a test that checks an upstream access log with retry (total of 4 logs - start, end repeated)